db/fixtures/ae_datastore/ManageIQ/Deployment/System/StateMachine/Deployment.class/__methods__/pre_deployment.rb

```diff
@@ -0,0 +1,93 @@
+require 'net/ssh'
+
+LOCAL_BOOK = 'local_book.yaml'.freeze
+
+def handle_rhel_subscriptions(commands)
+  system "scp -o 'StrictHostKeyChecking no' rhel_subscribe_inventory.yaml " \
+    "#{$evm.root['user']}@#{$evm.root['deployment_master']}:~/"
+  rhel_subscribe_cmd = "ansible-playbook /usr/share/ansible/openshift-ansible/playbooks/byo/rhel_subscribe.yml -i "\
+                       "/usr/share/ansible/openshift-ansible/rhel_subscribe_inventory.yaml"
+  commands.push("sudo mv ~/rhel_subscribe_inventory.yaml /usr/share/ansible/openshift-ansible/", rhel_subscribe_cmd)
+end
+
+def ssh_exec!(ssh, command)
+  stdout_data, stderr_data = "", ""
+  exit_code, exit_signal = nil, nil
+
+  ssh.open_channel do |channel|
+    channel.exec("ssh -A -o 'StrictHostKeyChecking no' -t -t #{$evm.root['user']}@#{$evm.root['deployment_master']} " \
+                 + command) do |_, success|
+      raise StandardError, "Command \"#{command}\" was unable to execute" unless success
+
+      channel.on_data do |_, data|
+        stdout_data += data
+      end
+
+      channel.on_extended_data do |_, _, data|
+        stderr_data += data
+      end
+
+      channel.on_request("exit-status") do |_, data|
+        exit_code = data.read_long
+      end
+
+      channel.on_request("exit-signal") do |_, data|
+        exit_signal = data.read_long
+      end
+    end
+  end
+  ssh.loop
+  {
+    :stdout      => stdout_data,
+    :stderr      => stderr_data,
+    :exit_code   => exit_code,
+    :exit_signal => exit_signal
+  }
+end
+
+def pre_deployment
+  commands = ['sudo yum install ansible -y',
+              'sudo yum install openshift-ansible -y',
+              'sudo yum install openshift-ansible-playbooks -y',
+              'sudo yum install pyOpenSSL -y',
```

Comment:
> @dkorn isn't:
> 
> sudo yum install -y openshift-ansible openshift-ansible-playbooks pyOpenSSL
> Faster?
> 

yes it is, I'll make the change

> @detiber what are the official packages that we need? For example I took out ansible because I think it's already required by openshift-ansible. Any reason to have to specify pyOpenSSL as well?

actually, IINM openshift-ansible-playbooks depends on ansible, openshift-ansible and pyOpenSSL. if that's true, would it be better to just execute:
sudo yum install -y openshift-ansible-playbooks?